overlord/snapstate/autorefresh.go

```diff
@@ -237,6 +238,35 @@ func (m *autoRefresh) Ensure() error {
 
 	// do refresh attempt (if needed)
 	if !m.nextRefresh.After(now) {
+		var can bool
+
+		can, err = canRefreshOnMeteredConnection(m.state)
+		if err != nil {
+			return err
+		}
+		if !can {
+			metered, _ := IsOnMeteredConnection()
+			if metered {
```

Comment:
I wonder if  this bit of checking and canRefreshOnMeteredConnection can be folded in one helper,

canRefreshRespectingMetered ?

Reply:
Thanks for the suggestion! The code got simplified nicely.